src/test/java/org/sonarlint/intellij/mediumtests/SecurityHotspotsMediumTest.kt

```diff
@@ -22,81 +22,188 @@ package org.sonarlint.intellij.mediumtests
+    }
+
+    @Test
+    fun should_match_security_hotspot_previously_detected_on_the_server() {
+        prepareStorageAndServer(
+            serverSecurityHotspot = ServerSecurityHotspot(
+                filePath = "file.rb",
+                key = "hotspotKey",
+                message = "Make sure using this hardcoded IP address is safe here.",
+                ruleKey = "ruby:S1313",
+                textRange = TextRange(1, 5, 1, 20),
+                introductionDate = "2020-09-21T12:46:39+0000",
+                status = "TO_REVIEW",
+                vulnerabilityProbability = "LOW"
+            )
+        )
 
-        assertThat(hotspots).isEmpty()
+        val raisedHotspot = analyzeFile(filePath = "file.rb", codeSnippet = "ip = \"192.168.12.42\";").first()
+
+        assertThat(raisedHotspot).extracting({ it.serverFindingKey },
+            { it.isResolved },
+            { it.introductionDate },
+            { it.vulnerabilityProbability }).containsExactly("hotspotKey", false, 1600692399000L, VulnerabilityProbability.LOW)
+    }
+
+    private fun prepareStorageAndServer(
+        serverSecurityHotspot: ServerSecurityHotspot,
+        branchName: String = "master",
+    ) {
+        createStorage(activeRuleKey = serverSecurityHotspot.ruleKey, projectKey = serverSecurityHotspot.projectKey, branchName = branchName)
+        mockServer.addProtobufResponse(
+            "/api/hotspots/search.protobuf?projectKey=${serverSecurityHotspot.projectKey}&files=${serverSecurityHotspot.filePath}&branch=$branchName&ps=500&p=1",
+            Hotspots.SearchWsResponse.newBuilder().setPaging(Common.Paging.newBuilder().setTotal(1).build()).addHotspots(
+                Hotspots.SearchWsResponse.Hotspot.newBuilder().setKey(serverSecurityHotspot.key).setMessage(serverSecurityHotspot.message)
+                    .setRuleKey(serverSecurityHotspot.ruleKey).setStatus(serverSecurityHotspot.status)
+                    .setVulnerabilityProbability(serverSecurityHotspot.vulnerabilityProbability)
+                    .setCreationDate(serverSecurityHotspot.introductionDate).setComponent(serverSecurityHotspot.componentKey).setTextRange(
+                        Common.TextRange.newBuilder().setStartLine(serverSecurityHotspot.textRange.startLine)
+                            .setStartOffset(serverSecurityHotspot.textRange.startLineOffset)
+                            .setEndLine(serverSecurityHotspot.textRange.endLine).setEndOffset(serverSecurityHotspot.textRange.endLineOffset)
+                            .build()
+                    ).build()
+            ).addComponents(
+                Hotspots.Component.newBuilder().setKey(serverSecurityHotspot.componentKey).setPath(serverSecurityHotspot.filePath).build()
+            ).build()
+        )
     }
 
     private fun createStorage(
-        serverVersion: String? = "9.7",
-        activeRuleKey: String
+        serverVersion: String? = "99.9",
```

Comment:
99.9 on purpose? or typo error for 9.9?

Reply:
it's on purpose, I want the default value to be something high. When not providing the argument in the call from the test, it means you want to simulate a recent server version